src/frontends/tensorflow/src/op/slice.cpp

```diff
@@ -18,15 +18,45 @@ OutputVector translate_slice_op(const NodeContext& node) {
     auto start = node.get_input(1);
     auto size = node.get_input(2);
 
-    auto stop = make_shared<Add>(start, size);
+    auto size_const_in = get_constant_from_source(size);
+    const auto& size_vector = size_const_in->cast_vector<int64_t>();
+
+    auto start_const_in = get_constant_from_source(start);
+    const auto& start_vector = start_const_in->cast_vector<int64_t>();
+
+    std::vector<int64_t> stop_vector(size_vector.size());
+
+    if (size_vector.size() != input.get_shape().size()) {
+        FRONT_END_THROW("Slice size vector length is not equal to number of input dimensions.");
+    }
+
+    // Size value -1 => Slice from the "start" value to the end
+    for (int i=0; i<size_vector.size(); i++) {
+        if (size_vector[i] == -1) {
+            stop_vector[i] = input.get_shape()[i];
+        } else {
+            stop_vector[i] = start_vector[i] + size_vector[i];
+        }
+    }
```

Comment:
Please implement for the general case when `begin` and `size` to be dynamic. In this case you need to replace TF `Slice` with a sub-graph:
```
end = Add (begin, size)
result = Slice(begin, end)
```
Make a note that you need to add additional sub-graph that checks `size` is negative or possitive using a mask `Less(size,0)` based on which you will differently compute `end`